src/audio/pcm_converter/pcm_remap.c

```diff
@@ -0,0 +1,503 @@
+		     struct audio_stream *sink, uint32_t dummy2,
+		     uint32_t source_samples, uint32_t chmap)
+{
+	return remap_c32_left_shift(source, sink, source_samples, chmap, 0);
+}
+
+static int remap_c32_to_c16_right_shift_16(const struct audio_stream *source, uint32_t dummy1,
+					   struct audio_stream *sink, uint32_t dummy2,
+					   uint32_t source_samples, uint32_t chmap)
+{
+	return remap_c32_to_c16(source, sink, source_samples, chmap, 16);
+}
+
+static int remap_c16_to_c32_left_shift_16(const struct audio_stream *source, uint32_t dummy1,
+					  struct audio_stream *sink, uint32_t dummy2,
+					  uint32_t source_samples, uint32_t chmap)
+{
+	return remap_c16_to_c32(source, sink, source_samples, chmap, 16);
+}
+
+static int remap_c32_to_c16_right_shift_8(const struct audio_stream *source, uint32_t dummy1,
+					  struct audio_stream *sink, uint32_t dummy2,
+					  uint32_t source_samples, uint32_t chmap)
+{
+	return remap_c32_to_c16(source, sink, source_samples, chmap, 8);
+}
+
+static int remap_c16_to_c32_left_shift_8(const struct audio_stream *source, uint32_t dummy1,
+					 struct audio_stream *sink, uint32_t dummy2,
+					 uint32_t source_samples, uint32_t chmap)
+{
+	return remap_c16_to_c32(source, sink, source_samples, chmap, 8);
+}
+
+static int remap_c32_right_shift_8(const struct audio_stream *source, uint32_t dummy1,
+				   struct audio_stream *sink, uint32_t dummy2,
+				   uint32_t source_samples, uint32_t chmap)
+{
+	return remap_c32_right_shift(source, sink, source_samples, chmap, 8);
+}
+
+static int remap_c32_left_shift_8(const struct audio_stream *source, uint32_t dummy1,
+				  struct audio_stream *sink, uint32_t dummy2,
+				  uint32_t source_samples, uint32_t chmap)
+{
+	return remap_c32_left_shift(source, sink, source_samples, chmap, 8);
+}
+
+static int remap_c32_right_shift_16(const struct audio_stream *source, uint32_t dummy1,
+				    struct audio_stream *sink, uint32_t dummy2,
+				    uint32_t source_samples, uint32_t chmap)
+{
+	return remap_c32_right_shift(source, sink, source_samples, chmap, 16);
+}
+
+static int remap_c32_left_shift_16(const struct audio_stream *source, uint32_t dummy1,
+				   struct audio_stream *sink, uint32_t dummy2,
+				   uint32_t source_samples, uint32_t chmap)
+{
+	return remap_c32_left_shift(source, sink, source_samples, chmap, 16);
+}
+
+static int remap_c32_to_c16_no_shift(const struct audio_stream *source, uint32_t dummy1,
+				     struct audio_stream *sink, uint32_t dummy2,
+				     uint32_t source_samples, uint32_t chmap)
+{
+	return remap_c32_to_c16(source, sink, source_samples, chmap, 0);
+}
+
+static int remap_c16_to_c32_no_shift(const struct audio_stream *source, uint32_t dummy1,
+				     struct audio_stream *sink, uint32_t dummy2,
+				     uint32_t source_samples, uint32_t chmap)
+{
+	return remap_c16_to_c32(source, sink, source_samples, chmap, 0);
+}
+
+/* Unfortunately, all these nice "if"s were commented out to suppress
+ * CI "defined but not used" warnings.
+ */
```

Comment:
That is not that easy. For example, for remap_c16_to_c32() it would be:
```
#if CONFIG_PCM_CONVERTER_FORMAT_S16LE && (CONFIG_PCM_CONVERTER_FORMAT_S32LE || CONFIG_PCM_CONVERTER_FORMAT_S24_4LE || CONFIG_PCM_CONVERTER_FORMAT_S24_4LE_MSB || CONFIG_PCM_CONVERTER_FORMAT_S16_4LE)
```

Reply:
yeah, well, it's nothing critical - just thinking out loud. You could define some intermediate flags like
```
#define X_IS_NEEDED (CONFIG_X1 || CONFIG_X2...)
```
but doesn't seem too important to me, up to you basically